### PR TITLE
Added methods to find artefact by both logical property name and namespace

### PR DESCRIPTION
**Pull request by davydotcom**
This adds methods to `GrailsApplicaiton` and `ArtefactInfo` interfaces.

### CODE REVIEW

**Comment by davydotcom** [comment]
weird not sure how i missed that one MockGrailsApplication will fix.

**Comment by graemerocher** [comment]
@davydotcom You still planning on correcting the pull request?

**Comment by davydotcom** [comment]
Yes I was on vacation last week

File: grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/ArtefactInfo.java [hunk 1]
```diff
@@ -70,4 +71,21 @@
...
     GrailsClass getGrailsClassByLogicalPropertyName(String logicalName);
+
+    /**
+    * Retrieves a list of Grails classes by their logical property name. This can grab multiple classes that have
+    * the same name but may have different namespaces.
+    *
+    * @param logicalName The logical name
+    * @return A List<GrailsClass> or null
+    */
+    LinkedList<GrailsClass> getGrailsClassesByLogicalPropertyName(String logicalName);
```
Comment (driverpt) [inline grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/ArtefactInfo.java]: LinkedList ?

File: grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultArtefactInfo.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
     private Class<?>[] classes;
     private Map<String, GrailsClass> grailsClassesByName = new LinkedHashMap<String, GrailsClass>();
     private Map<String, Class<?>> classesByName = new LinkedHashMap<String, Class<?>>();
-    private Map<String, GrailsClass> logicalPropertyNameToClassMap = new HashMap<String, GrailsClass>();
+    private Map<String, LinkedList<GrailsClass>> logicalPropertyNameToClassMap = new HashMap<String, LinkedList<GrailsClass>>();
```
Comment (driverpt) [inline grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultArtefactInfo.java]: Why a LinkedList and not a List ?

File: grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultArtefactInfo.java [hunk 3]
```diff
@@ -61,7 +61,16 @@ private void addGrailsClassInternal(GrailsClass artefactClass, boolean atStart)
             grailsClasses.remove(oldVersion);
         }
         classesByName.put(actualClass.getName(), actualClass);
-        logicalPropertyNameToClassMap.put(artefactClass.getLogicalPropertyName(), artefactClass);
+
+        LinkedList<GrailsClass> grailsClassesByLogicalPropertyName = logicalPropertyNameToClassMap.get(artefactClass.getLogicalPropertyName());
```
Comment (driverpt) [inline grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/DefaultArtefactInfo.java]: Same as above

**Comment by driverpt** [comment]
I recommend changing from LinkedList to List.

**Comment by davydotcom** [comment]
this is old, probably not valid anymore...closing

**Pull request closed without merging**